### PR TITLE
models inputs, outputs correlation matrix

### PR DESCRIPTION
**Pull request by nden**
This modifies `is_separable` to return the correlation matrix of inputs and outputs. 
Motivation for this is the WCS APE.

### CODE REVIEW

**Comment by pllim** [comment]
This is an API change, why not milestone to 3.1? Is there a need to back port?

File: CHANGES.rst [hunk 1]
```diff
@@ -579,6 +579,10 @@ astropy.wcs
 Other Changes and Additions
 ---------------------------
 
+astropy.modeling
+
+- ``is_separable`` not returns the correlation matrix of inputs
+  and outputs.                                      [#7803]
```
[the remark this answers is not in the document]
Comment (astrofrog) [reply CHANGES.rst]: #teamocd

**Comment by nden** [comment]
Addressed comments and changed milestone to 3.1.

**Comment by astrofrog** [comment]
Is there a reason we can't simply add a new property ``separability_matrix`` rather than modifying the existing ``is_separable``? In addition to being simpler for users to specify what exactly they want, it has the benefit of being backward-compatible.

**Comment by nden** [comment]
@astrofrog Yes, separating the two functionalities would be a good solution. 
If it's a property of the `Model` class the contents of `separable.py` needs to be merged with core.py making it even longer. Is keeping it as a function in `separable.p` less intuitive?

**Comment by astrofrog** [comment]
Ah sorry I didn't realize it was in a separate file - in that case a separate function in ``separable.py`` would be fine.

**Comment by Cadair** [comment]
+1 for having two different things. Having to extract the diagonal of a matrix to get back the current behavior would be irritating.

**Comment by nden** [comment]
I refactored the functionality into a separate function. This is now ready for review.

**Pull request merged**